frontend: hide managed fields

farodin91 opened the pull request.

farodin91 commented:
I had this idea from argo-cd.

ashu8912 reviewed (approved):
Tested, LGTM!!

sniok commented on frontend/src/components/common/Resource/EditorDialog.tsx:
```diff
@@ -109,6 +112,15 @@ export default function EditorDialog(props: EditorDialogProps) {
 
   const theme = useCurrentAppTheme();
 
+  const [hideManagedFields, setHideManagedFieldsState] = React.useState(() => {
```
we have a `useLocalStorageState` hook that you could use here instead

farodin91 replied:
used.

farodin91 commented on frontend/src/components/common/Resource/EditorDialog.tsx:
```diff
@@ -444,6 +463,20 @@ export default function EditorDialog(props: EditorDialogProps) {
+                        label={t('Hide Managed Fields')}
+                      />
+                    ) : (
+                      <div />
```
removed

sniok commented on frontend/src/components/common/Resource/EditorDialog.tsx:
```diff
@@ -109,34 +113,41 @@ export default function EditorDialog(props: EditorDialogProps) {
-  });
+  const [hideManagedFields, setHideManagedFields] = useLocalStorageState<boolean>(
+    'hideManagedFields',
+    false
```
I'd make the default setting here "true", that's what kubectl does by default and I think we should too. But this is not too critical since you only have to check it once and it'll remember it.
What do you think @farodin91 @joaquimrocha ?

joaquimrocha replied:
Yeah, let's turn it on by default please.

farodin91 commented:
Rebased and fixed. 

Another idea:
I tried shortly to use folding section in monaco but didn't get it to work. Basic always folding the managedfields section in the editor. If anyone thinks it is a better idea and get it work go for it.

sniok reviewed (approved):
nice, thanks! I think folding would be super cool but from what I remember it's quite tricky to implement so let's keep it simple

illume commented:
Very cool. Thanks.

btw. We don’t need Signed-off-by anymore now that we’re using the EasyCLA thing.
